Makefile

```diff
@@ -100,23 +100,24 @@ GO_TEST_FORMAT = pkgname
 DEFAULT_RELEASE_IMAGES = $(shell (tr -d '\n\t ' < ${ROOT_DIR}/data/default_release_images.json))
 DEFAULT_OS_IMAGES = $(shell (tr -d '\n\t ' < ${ROOT_DIR}/data/default_os_images.json))
 
+# Support all Release/OS images for CI
 ifeq ($(CI), true)
 	VERBOSE = true
-	OPENSHIFT_VERSIONS := $(or ${OPENSHIFT_VERSIONS}, $(shell hack/get_ocp_versions_for_testing.sh))
-else
-	OPENSHIFT_VERSIONS := $(or ${OPENSHIFT_VERSIONS}, $(shell hack/get_ocp_versions_for_testing.sh | jq -c 'with_entries(select(.value.default))'))
-	ifneq ($(OPENSHIFT_VERSIONS), {})
-		DEFAULT_VERSION := $(shell (jq -n '$(OPENSHIFT_VERSIONS)' | jq 'keys[0]'))
-		RELEASE_IMAGES := $(or ${RELEASE_IMAGES}, $(shell (echo '$(DEFAULT_RELEASE_IMAGES)' | jq -c --arg v $(DEFAULT_VERSION) 'map(select(.openshift_version==$$v))')))
-		OS_IMAGES := $(or ${OS_IMAGES}, $(shell (echo '$(DEFAULT_OS_IMAGES)' | jq -c --arg v $(DEFAULT_VERSION) 'map(select(.openshift_version==$$v))')))
-	endif
+	RELEASE_IMAGES := $(or ${RELEASE_IMAGES},${DEFAULT_RELEASE_IMAGES})
+	OS_IMAGES := $(or ${OS_IMAGES},${DEFAULT_OS_IMAGES})
 endif
 
-RELEASE_IMAGES := $(or ${RELEASE_IMAGES},${DEFAULT_RELEASE_IMAGES})
-OS_IMAGES := $(or ${OS_IMAGES},${DEFAULT_OS_IMAGES})
+# Support default Release/OS image if lists not specified
+ifndef RELEASE_IMAGES
+ifndef OS_IMAGES
+	DEFAULT_VERSION := $(shell (echo '$(DEFAULT_RELEASE_IMAGES)' | jq -c 'map(select(.default))[0].openshift_version'))
+	RELEASE_IMAGES := $(shell (echo '$(DEFAULT_RELEASE_IMAGES)' | jq -c --arg v $(DEFAULT_VERSION) 'map(select(.openshift_version==$$v))'))
+	OS_IMAGES := $(shell (echo '$(DEFAULT_OS_IMAGES)' | jq -c --arg v $(DEFAULT_VERSION) 'map(select(.openshift_version==$$v))'))
+endif
+endif
 
-# RHCOS_VERSION should be consistent with BaseObjectName in pkg/s3wrapper/client.go
-RHCOS_BASE_ISO := $(shell (jq -n '$(OPENSHIFT_VERSIONS)' | jq '[.[].rhcos_image]|max'))
+# Support all OS images if missing - for kube-api flow
+OS_IMAGES := $(or ${OS_IMAGES},${DEFAULT_OS_IMAGES})
```

Comment:
It means that every deployment from test-infra will deploy all the versions, if I'm not mistaken (since test-infra overrides the environment variables)
So it needs to come up with a solution to a local test-infra (not CI) that deploys a single image to reduce deployment time

Reply:
Isn't this line a duplication of line 107?

Reply:
Nope, since line 107 is only for CI=true